pandas/core/indexes/multi.py

```diff
@@ -1412,6 +1420,107 @@ def from_product(cls, iterables, sortorder=None, names=None):
         labels = cartesian_product(labels)
         return MultiIndex(levels, labels, sortorder=sortorder, names=names)
 
+    @classmethod
+    def from_frame(cls, df, sortorder=None, names=None, squeeze=True):
+        """
+        Make a MultiIndex from a DataFrame.
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            DataFrame to be converted to MultiIndex.
+        sortorder : int or None
+            Level of sortedness (must be lexicographically sorted by that
+            level).
+        names : list-like / callable, optonal
+            If no names provided, use column names, or tuple of column names if
+            the columns is a MultiIndex. If sequence, overwrite names with the
+            given sequence. If callable, pass each column name or tuples of
+            names to the callable.
+        squeeze : bool, default True
+            If df is a single column, squeeze MultiIndex to be a regular Index.
+
+        Returns
+        -------
+        MultiIndex or Index
+            The MultiIndex representation of the given DataFrame. Returns an
+            Index if the DataFrame is single column and squeeze is True.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame([[0, 'happy'], [0, 'jolly'], [1, 'happy'],
+        ...                    [1, 'jolly'], [2, 'joy'], [2, 'joy']],
+        ...                   columns=['number', 'mood'])
+        >>> df
+           number   mood
+        0       0  happy
+        1       0  jolly
+        2       1  happy
+        3       1  jolly
+        4       2    joy
+        5       2    joy
+        >>> pd.MultiIndex.from_frame(df)
+        MultiIndex(levels=[[0, 1, 2], ['happy', 'jolly', 'joy']],
+                   labels=[[0, 0, 1, 1, 2, 2], [0, 1, 0, 1, 2, 2]],
+                   names=['number', 'mood'])
+        >>> df = pd.DataFrame([[0], [1]], columns=['number'])
+        >>> df
+           number
+        0       0
+        1       1
+        >>> pd.MultiIndex.from_frame(df)
```

Comment:
can you put a blank line between cases

Reply:
adding a comment on the case if its not obvious

Reply:
removed other examples as I have removed the callable names /squeeze parameter